hatch.toml

```diff
@@ -19,14 +19,17 @@ features = ["test", "dask-ml"]
 extra-dependencies = ["ipykernel"]
 overrides.matrix.deps.env-vars = [
     { if = ["pre"], key = "UV_PRERELEASE", value = "allow" },
-    { if = ["min"], key = "UV_RESOLUTION", value = "lowest-direct" },
+    { if = ["min"], key = "UV_CONSTRAINT", value = "ci/scanpy-min-deps.txt" },
+]
+overrides.matrix.deps.pre-install-commands = [
+    { if = ["min"], value = "uv pip install -c '' tomli packaging && python ci/scripts/min-deps.py pyproject.toml -o ci/scanpy-min-deps.txt" },
```

Comment:
Does installing `tomli` and `packaging` here potentially affect the CI env? 
Maybe this would be a good fit for using `uv run ci/scripts/min-deps.py` and declare the dependencies [within the script](https://docs.astral.sh/uv/guides/scripts/#running-a-script-with-dependencies)?

Reply:
that’s a great point! I’ll do that